ci(issue-23) - Multiple OS Binaries

fras2560 opened the pull request:
Extend the existing Github Action to allow for builds of a given OS and Arch
Extend the dev and prod workflows to publish binaries for Darwin + Windows
Closes Issue-23 for Canvas-Desktop

fras2560 committed:
ci(issue-23): Refactor build action to support different os and arch

fras2560 committed:
ci: build and publish binaries for linux, MacOs and Windows

fras2560 committed:
missing inputs keyword

fras2560 committed:
inputs not input for github action

brandonbloch commented on .github/actions/build/action.yml:
```diff
@@ -1,13 +1,32 @@
+    required: false
+    default: amd64
+  build-suffix:
+    description: The suffic to add to the built binary and its artifact upload
```
Typo in "suffix"

brandonbloch commented on .github/actions/build/action.yml:
```diff
@@ -1,13 +1,32 @@
-        path: stl-to-3mf
+        name: stl-to-3mf${{inputs.build-suffix}}
+        path: |
+          stl-to-3mf${{inputs.build-suffix}}
+          stl-to-3mf${{inputs.build-suffix}}.exe
```
Should this just be `stl-to-3mf${{inputs.build-suffix}}`? Since only windows builds should have a `.exe` file

brandonbloch commented on .github/workflows/dev.yml:
```diff
@@ -22,6 +22,13 @@ jobs:
       - uses: ./.github/actions/build
+      - uses: ./.github/actions/build
+        with:
+          os: windows
```
Should the suffix for Windows be `.exe`?

brandonbloch commented on .github/workflows/prod.yml:
```diff
@@ -22,6 +22,13 @@ jobs:
       - uses: ./.github/actions/build
+      - uses: ./.github/actions/build
+        with:
+          os: windows
```
Same here re. `.exe` suffix

fras2560 committed:
change .exe as a build suffix

fix typo in description

fras2560 committed:
ci: upload binary versions for all os

Update PR workflow to check if a chance was made to version.txt
Add a readme that mentions how versioning works

fras2560 committed:
remove extra bracket

fras2560 committed:
remove extra } on check
